nomad/vault.go

```diff
@@ -1022,7 +1033,14 @@ func (v *vaultClient) CreateToken(ctx context.Context, a *structs.Allocation, ta
 		return nil, err
 	}
 
-	if v.tokenData.Root && role == "" {
+	noRole := v.tokenData.Root && role == ""
+
+	if noRole && req.EntityAlias != "" {
+		err = fmt.Errorf("Entity Alias must be used with a Vault token that has a role")
+		return nil, err
+	}
```

Comment:
Question on 1038-1041, If the user passes in an entity-alias and is not using a role, should we error from Nomad (as shown here), or let Vault error?

In a different but similar situation, if the user has an invalid entity-alias (which Vault knows but Nomad doesn't), we still make the call to Vault and then just surface that error. We could just let the user make the bad call to Vault in this case too, as the error message in that case is pretty clear. Thoughts?